File: leetcode/kotlin/11.ContainerWithMostMediumWater.kt
import kotlin.math.min

class Solution{
    companion object{
        fun maxArea(height: List<Int>):Int{
          var max:Int=0
          var left:Int=0
          var right=height.size-1
          while(left<right){
              max = kotlin.math.max(max, min(height[left], height[right]) * (right - left));
              if (height[left] < height[right]) {
                  left ++;
              }
              else {
                  right --;
              }
          }
          return max
        }
    }
}

fun main(){
    val arr= listOf<Int>(1,2,1)
    print(Solution.maxArea(arr))
}

File: leetcode/c++/11.ContainerWithMostMediumWater.cpp
#include <iostream>
#include <algorithm> 
#include <vector>

using namespace std;

class Solution{
  public:
  static int maxArea(vector<int>& height){
    int left{0};
    int right= height.size()-1;
    int max=0;
    while (left < right) {
      max = std::max(max,min(height[left], height[right]) * (right - left));
      if (height[left] < height[right]) {
        left ++;
      }
      else {
        right --;
      }
    }
    return max;
  }
};

int main(){
  vector<int> arr{1,2,1};
  cout<<Solution::maxArea(arr)<<endl;
  return 0;
}